### PR TITLE
Update infrastructure

### LINKED ISSUE #794: Update infrastructure
- [x] conda -> pixi
- [x] setup.py -> hatchling
- [x] make sure rtd still works (at present, no separate environment file)
- [x] update hooks to latest template 
- [x] check pyproject.toml contents (careful with docstring style)
- [x] add pixi section to developer docs

### PR DESCRIPTION
**Pull request by timmens**
Closes #794 

### ToDo

- [x] Rename `environment.yml` to `rtd_environment.yml` on RTD
- [x] Check whether codecov failure is relevant
- [ ] ~~Check why I need to run `dot -c` on the macOS runners such that pygraphviz dot renderer is properly configured and why this happens now~~

### CODE REVIEW

File: .github/workflows/main.yaml [hunk 1]
```diff
@@ -22,58 +22,47 @@ jobs:
...
-          init-shell: bash
-          environment-file: environment.yml
-          environment-name: gettsim
-          cache-environment: true
-          create-args: >-
-            python=${{ matrix.python-version }}
-            jax
-      - name: Run tests with NumPy backend on Linux
-        # Ignore MacOS for being slow -- no reason that would break if Jax works.
+          pixi-version: v0.29.0
```
Comment (hmgaudecker) [inline .github/workflows/main.yaml:44]: Now at 0.34 -- any way to auto-update this?

File: pyproject.toml [hunk 2]
```diff
@@ -187,13 +291,21 @@ ignore = [
 ]
 
 
+# ======================================================================================
```
Comment (hmgaudecker) [inline pyproject.toml:294]: This section can be safely removed, right? All done by ruff if I am not mistaken.

**Review by hmgaudecker — APPROVED** [review]
Very nice and helpful, thanks a lot! 

If you don't mind, could you take this occasion to update the badges in the README file?

File: .github/workflows/main.yaml (hunk 1 above)
Comment (timmens) [reply .github/workflows/main.yaml:44]: I could not find anything, but only looked for a brief moment.

**Review by hmgaudecker — APPROVED** [review]
Thanks!!! Tiny thing: Could we replace the anaconda badges by conda-forge and pypi? For sure the version on anaconda is outdated. We'll probably want to delete the gettsim channel at some point.

**Pull request merged**